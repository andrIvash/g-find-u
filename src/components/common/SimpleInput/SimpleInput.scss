@import '../../../styles/abstracts/functions';

$input-min-width: rem(140);

.simpleInput {
    input {
        min-width: $input-min-width;
    }
}