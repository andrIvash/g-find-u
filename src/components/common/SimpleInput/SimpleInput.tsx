import * as React from 'react';
import TextField from '@mui/material/TextField';
import { Props, defaultLabel } from './index';

import './SimpleInput.scss';

export const SimpleInput = (props: Props) => {
    
    const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
        props.onChange(event.target.value)
    };

    return (
        <div
            className={`simpleInput input ${props.className ? props.className : ""}`}
            data-testid='SimpleInput'
        >
            <TextField
                className="textField"
                id="outlined-controlled"
                label={props.label ? props.label : defaultLabel}
                value={props.value}
                onChange={handleChange}
            />
        </div>
    )
}
