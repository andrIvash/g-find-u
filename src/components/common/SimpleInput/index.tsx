import { SimpleInput } from './SimpleInput';

export const defaultLabel = 'Search';

export interface Props {
    onChange: (value: string) => void,
    value: string,
    label?: string,
    className?: string
}

export default SimpleInput;