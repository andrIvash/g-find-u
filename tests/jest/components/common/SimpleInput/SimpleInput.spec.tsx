import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom';
import SimpleInput from '../../../../../src/components/common/SimpleInput';

describe('SimpleInput Component', () => {
    const mockOnChange = jest.fn();
    it('renders without crashing', () => {
        render(<SimpleInput onChange={mockOnChange} value='' />);
        expect(screen.getByTestId('SimpleInput')).toBeInTheDocument();
    });

    it('handles className prop', () => {
        render(<SimpleInput onChange={mockOnChange} value='' className='testClass' />);
        expect(screen.getByTestId('SimpleInput')).toHaveClass('testClass');
    });

    it('uses default label when label prop is not provided', () => {
        const defaultLabelText = 'Search';
        render(<SimpleInput onChange={mockOnChange} value='' />);
        expect(screen.getByLabelText(defaultLabelText)).toBeInTheDocument();
    });

    it('calls onChange prop when changed', () => {
        render(<SimpleInput onChange={mockOnChange} value='' />);
        fireEvent.change(screen.getByRole('textbox'), { target: { value: 'New Value' } });
        expect(mockOnChange).toHaveBeenCalledTimes(1);
    });

    it('matches snapshot', () => {
        const { asFragment } = render(<SimpleInput onChange={mockOnChange} value='' />);
        expect(asFragment()).toMatchSnapshot();
    });
});