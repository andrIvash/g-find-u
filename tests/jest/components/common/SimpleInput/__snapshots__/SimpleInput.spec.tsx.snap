// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SimpleInput Component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="simpleInput input "
    data-testid="SimpleInput"
  >
    <div
      class="MuiFormControl-root MuiTextField-root textField css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
    >
      <label
        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink="false"
        for="outlined-controlled"
        id="outlined-controlled-label"
      >
        Search
      </label>
      <div
        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
      >
        <input
          aria-invalid="false"
          class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
          id="outlined-controlled"
          type="text"
          value=""
        />
        <fieldset
          aria-hidden="true"
          class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="css-yjsfm1"
          >
            <span>
              Search
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
